tests/test_codex_feed_hooks.py

```
@@ -78,6 +103,289 @@ def _run(self) -> None:
+                check=False,
+                env=env,
+                timeout=10,
+            )
+            if result.returncode != 0:
+                raise AssertionError(
+                    f"hooks codex stop failed exit={result.returncode}\n"
+                    f"stdout={result.stdout}\nstderr={result.stderr}"
+                )
+            time.sleep(0.5)
+            if not monitor_pids_for_session(session_id):
+                raise AssertionError("turn-less Stop reaped a session-wide monitor")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)